--- kernel/drivers/net/wireless/bcm4329/dhd.h.orig	2012-05-31 08:58:04.000000000 +0200
+++ kernel/drivers/net/wireless/bcm4329/dhd.h	2012-05-31 08:57:10.000000000 +0200
@@ -1,3 +1,5 @@
+#pragma GCC diagnostic warning "-Wunused-but-set-variable"
+#pragma GCC diagnostic warning "-Warray-bounds"
 /*
  * Header file describing the internal (inter-module) DHD interfaces.
  *
--- kernel/drivers/net/wireless/bcm4329/siutils_priv.h	2012-05-31 08:57:01.000000000 +0200
+++ kernel/drivers/net/wireless/bcm4329/siutils_priv.h.orig	2012-05-31 09:00:47.000000000 +0200
@@ -1,3 +1,4 @@
+#pragma GCC diagnostic warning "-Wunused-but-set-variable"
 /*
  * Include file private to the SOC Interconnect support files.
  *
--- kernel/drivers/net/wireless/bcm4329/dhd_linux.c.orig	2011-12-21 19:54:24.884526348 +0100
+++ kernel/drivers/net/wireless/bcm4329/dhd_linux.c	2011-12-21 18:55:18.974476355 +0100
@@ -1904,6 +1904,13 @@ dhd_open(struct net_device *net)
 	ifidx = dhd_net2idx(dhd, net);
 	DHD_TRACE(("%s: ifidx %d\n", __FUNCTION__, ifidx));
 
+//max
+	if (ifidx < 0 || ifidx >= DHD_MAX_IFS) {
+		DHD_ERROR(("%s: Error: error ifidx out of range\n", __FUNCTION__));
+		return -1;
+	}
+//end max
+
 	if ((dhd->iflist[ifidx]) && (dhd->iflist[ifidx]->state == WLC_E_IF_DEL)) {
 		DHD_ERROR(("%s: Error: called when IF already deleted\n", __FUNCTION__));
 		return -1;
