diff -Naur a/omx/gstomxvideoenc.c b/omx/gstomxvideoenc.c
--- a/omx/gstomxvideoenc.c	2016-06-08 06:17:32.000000000 +0200
+++ b/omx/gstomxvideoenc.c	2017-01-08 23:54:18.217242233 +0100
@@ -2440,7 +2440,7 @@
 
   err =
       gst_omx_component_get_index (self->enc,
-      NVX_INDEX_CONFIG_VIDEO_ENCODE_TEMPORALTRADEOFF, &eIndex);
+      (char *) NVX_INDEX_CONFIG_VIDEO_ENCODE_TEMPORALTRADEOFF, &eIndex);
   if (err != OMX_ErrorNone) {
     return FALSE;
   }
