From b18901617717029cbe26ecb578a6e9c5ec92d507 Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.krummenacher@toradex.com>
Date: Thu, 28 Sep 2017 15:05:35 +0200
Subject: [PATCH] configure: don't use gst-photography even if it is installed
 in the rootfs

Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 configure.ac | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 885720d..9aa608e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -275,11 +275,12 @@ AM_CONDITIONAL([GST_ENABLE_PHOTOGRAPHY], [test x$GST_ENABLE_PHOTOGRAPHY = xyes])
 dnl Check for GStreamer-Interfaces
 PKG_CHECK_MODULES([GST_INTERFACES], [gstreamer-interfaces-$GST_MAJORMINOR], HAVE_GST_INTERFACES=yes, HAVE_GST_INTERFACES=no)
 if test "x$HAVE_GST_INTERFACES" = xyes; then
-  PKG_CHECK_MODULES([GST_PHOTOGRAPHY], [gstreamer-plugins-bad-$GST_MAJORMINOR] >= $GST_PHOTOGRAPHY_REQUIRED, HAVE_GST_PHOTOGRAPHY=yes, HAVE_GST_PHOTOGRAPHY=no)
+  PKG_CHECK_MODULES([GST_PHOTOGRAPHY], [gstreamer-plugins-bad-$GST_MAJORMINOR] >= $GST_PHOTOGRAPHY_REQUIRED, HAVE_GST_PHOTOGRAPHY=no, HAVE_GST_PHOTOGRAPHY=no)
   if test "x$HAVE_GST_PHOTOGRAPHY" = xyes; then
-    GST_ENABLE_PHOTOGRAPHY=yes
+    dnl don't configure it, even if there is a gst-photography
+    GST_ENABLE_PHOTOGRAPHY=no
     AM_CONDITIONAL([GST_ENABLE_PHOTOGRAPHY], [test x$GST_ENABLE_PHOTOGRAPHY = xyes])
-    XOPT_CFLAGS+="  -DGST_ENABLE_PHOTOGRAPHY -DGST_USE_UNSTABLE_API"
+#    XOPT_CFLAGS+="  -DGST_ENABLE_PHOTOGRAPHY -DGST_USE_UNSTABLE_API"
   fi
 fi
 
-- 
2.9.4

