From 1306eabf258f0e9ce8586b7816406a21683d529d Mon Sep 17 00:00:00 2001
From: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Date: Wed, 8 Jun 2016 14:23:29 +0200
Subject: [PATCH] Fix cross compilation

Signed-off-by: Marcel Ziswiler <marcel.ziswiler@toradex.com>
---
 GNUmakefile | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index 3bc76a4..f71d990 100755
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -73,21 +73,6 @@ ifneq ($(IS_X86_64),0)
  endif
 endif
 
-# Guard use of -march=native
-ifeq ($(GCC_COMPILER),0)
-   CXXFLAGS += -march=native
-else ifneq ($(GCC42_OR_LATER),0)
-   CXXFLAGS += -march=native
-else
-  # GCC 3.3 and "unknown option -march="
-  # GCC 4.1 compiler crash with -march=native.
-  ifneq ($(IS_X86_64),0)
-    CXXFLAGS += -m64
-  else
-    CXXFLAGS += -m32
-  endif # X86/X32/X64
-endif
-
 # Aligned access required at -O3 for GCC due to vectorization (circa 08/2008). Expect other compilers to do the same.
 UNALIGNED_ACCESS := $(shell $(EGREP) -c "^[[:space:]]*//[[:space:]]*\#[[:space:]]*define[[:space:]]*CRYPTOPP_NO_UNALIGNED_DATA_ACCESS" config.h)
 ifeq ($(findstring -O3,$(CXXFLAGS)),-O3)
-- 
2.5.5

